src/OrchardCore.Modules/OrchardCore.Liquid/ScriptsMiddleware.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement.Liquid;
+using OrchardCore.Entities;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Settings;
+using Fluid;
+
+namespace OrchardCore.Liquid
+{
+    public class ScriptsMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ScriptsMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments("/OrchardCore.Liquid/Scripts", StringComparison.OrdinalIgnoreCase))
+            {
+                var script = default(string);
+                if (Path.GetFileName(httpContext.Request.Path.Value) == "liquid-intellisense.js")
+                {
+                    var templateOptions = httpContext.RequestServices.GetRequiredService<IOptions<TemplateOptions>>();
+
+                    var filters = string.Join(',', templateOptions.Value.Filters.Select(x => $"'{x.Key}'"));
+                    
+                    script = $@"[{filters}].forEach(value=>{{if(!liquidFilters.includes(value)){{ liquidFilters.push(value);}}}});";
```

Comment:
Would there be a way to add the dynamic tags too? Or is that you didn't find a way to list them? I might be able to help then.

Would awesome also if we could list the arguments, or a description for the filters/tags, if we can add some metadata when they are registered (would need to change fluid for that, or an other options in orchard).

Reply:
I couldn't find a way to list the tags, that's why i added only the filters.
There are two fields in suggestions, detail(string) and documentation([monaco.IMarkdownString](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.imarkdownstring.html)) which we could use.
The output then would be like 
![image](https://user-images.githubusercontent.com/9623255/120707498-0ff03780-c4c3-11eb-8eab-0432d69c7eb0.png)
collapsed, and expanded
![image](https://user-images.githubusercontent.com/9623255/120707544-226a7100-c4c3-11eb-985d-6e1a6ce82713.png)

I don't see any obstacle to add the arguments as suggestions (you mean ex. the User object in the above image? ) if we have a list of them.

Reply:
Try to resolve `LiquidViewParser`, then access the `RegisteredTags` that contains custom tags. If adding documentation requires more work then we can try to handle it in a separate PR.

Reply:
Thanks, that worked. Is there a way to know if a tag is a block and needs end tag? 
![image](https://user-images.githubusercontent.com/9623255/121060823-cdcd3b80-c7cb-11eb-9087-2f970f2c155b.png)
I agree to keep the documentation for another PR. We could also implement a completion provider for the shortcodes with documentation first.